backward.hpp

```diff
@@ -515,12 +587,20 @@ class StackTraceLinuxImplBase {
 
 protected:
 	void load_thread_info() {
+#ifdef BACKWARD_SYSTEM_LINUX
 		_thread_id = (size_t)syscall(SYS_gettid);
 		if (_thread_id == (size_t) getpid()) {
 			// If the thread is the main one, let's hide that.
 			// I like to keep little secret sometimes.
 			_thread_id = 0;
 		}
+#elif defined(BACKWARD_SYSTEM_DARWIN)
+		_thread_id = reinterpret_cast<size_t>(pthread_self());
```

Comment:
well pthread_t is opaque and there is no guarantee that you can convert it to a size_t. but for now I will accept this as it's not a regression.